Can't build mingw ruby 2.0.0-p0 on OS X

Running `make winruby` fails with this:

```
bundle exec rake-compiler cross-ruby VERSION=2.0.0-p0
cd /Users/ahellesoy/.rake-compiler/builds/ruby-2.0.0-p0
make
    CC = i686-w64-mingw32-gcc
    LD = i686-w64-mingw32-ld
    LDSHARED = i686-w64-mingw32-gcc -shared 
    CFLAGS = -O3 -fno-omit-frame-pointer -fno-fast-math -ggdb3 -Wall -Wextra -Wno-unused-parameter -Wno-parentheses -Wno-long-long -Wno-missing-field-initializers -Wunused-variable -Werror=pointer-arith -Werror=write-strings -Werror=declaration-after-statement -Werror=implicit-function-declaration  
    XCFLAGS = -include ruby/config.h -include ruby/missing.h -D_FORTIFY_SOURCE=2 -fno-strict-overflow -DRUBY_EXPORT
    CPPFLAGS = -D_WIN32_WINNT=0x0501   -I. -I.ext/include/i386-mingw32 -I/Users/ahellesoy/.rake-compiler/sources/ruby-2.0.0-p0/include -I/Users/ahellesoy/.rake-compiler/sources/ruby-2.0.0-p0
    DLDFLAGS = -Wl,--enable-auto-image-base,--enable-auto-import -Wl,--out-implib=libmsvcrt-ruby200.dll.a msvcrt-ruby200.def -Wl,--stack,0x00200000,--enable-auto-import  
    SOLIBS = msvcrt-ruby200.res.o -lshell32 -lws2_32 -limagehlp -lshlwapi  
compiling /Users/ahellesoy/.rake-compiler/sources/ruby-2.0.0-p0/dln.c
In file included from /Users/ahellesoy/.rake-compiler/sources/ruby-2.0.0-p0/include/ruby/defines.h:153:0,
                 from /Users/ahellesoy/.rake-compiler/sources/ruby-2.0.0-p0/include/ruby/ruby.h:70,
                 from /Users/ahellesoy/.rake-compiler/sources/ruby-2.0.0-p0/dln.c:13:
/Users/ahellesoy/.rake-compiler/sources/ruby-2.0.0-p0/include/ruby/win32.h: In function ‘rb_w32_pow’:
/Users/ahellesoy/.rake-compiler/sources/ruby-2.0.0-p0/include/ruby/win32.h:787:5: error: implicit declaration of function ‘_controlfp’
/Users/ahellesoy/.rake-compiler/sources/ruby-2.0.0-p0/include/ruby/win32.h:788:16: error: ‘_PC_64’ undeclared (first use in this function)
/Users/ahellesoy/.rake-compiler/sources/ruby-2.0.0-p0/include/ruby/win32.h:788:16: note: each undeclared identifier is reported only once for each function it appears in
/Users/ahellesoy/.rake-compiler/sources/ruby-2.0.0-p0/include/ruby/win32.h:788:24: error: ‘_MCW_PC’ undeclared (first use in this function)
make[1]: *** [dln.o] Error 1
/Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/file_utils.rb:53:in `block in create_shell_runner': Command failed with status (2): [make...] (RuntimeError)
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/file_utils.rb:45:in `call'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/file_utils.rb:45:in `sh'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/file_utils_ext.rb:40:in `sh'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0/gems/rake-compiler-0.8.3/tasks/bin/cross-ruby.rake:155:in `block (2 levels) in <top (required)>'
    from /Users/ahellesoy/.rvm/rubies/ruby-2.0.0-p0/lib/ruby/2.0.0/fileutils.rb:206:in `chdir'
    from /Users/ahellesoy/.rvm/rubies/ruby-2.0.0-p0/lib/ruby/2.0.0/fileutils.rb:206:in `cd'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/file_utils_ext.rb:40:in `chdir'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0/gems/rake-compiler-0.8.3/tasks/bin/cross-ruby.rake:154:in `block in <top (required)>'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/task.rb:226:in `call'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/task.rb:226:in `block in execute'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/task.rb:223:in `each'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/task.rb:223:in `execute'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/task.rb:166:in `block in invoke_with_call_chain'
    from /Users/ahellesoy/.rvm/rubies/ruby-2.0.0-p0/lib/ruby/2.0.0/monitor.rb:211:in `mon_synchronize'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/task.rb:159:in `invoke_with_call_chain'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/task.rb:187:in `block in invoke_prerequisites'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/task.rb:185:in `each'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/task.rb:185:in `invoke_prerequisites'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/task.rb:165:in `block in invoke_with_call_chain'
    from /Users/ahellesoy/.rvm/rubies/ruby-2.0.0-p0/lib/ruby/2.0.0/monitor.rb:211:in `mon_synchronize'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/task.rb:159:in `invoke_with_call_chain'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/task.rb:187:in `block in invoke_prerequisites'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/task.rb:185:in `each'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/task.rb:185:in `invoke_prerequisites'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/task.rb:165:in `block in invoke_with_call_chain'
    from /Users/ahellesoy/.rvm/rubies/ruby-2.0.0-p0/lib/ruby/2.0.0/monitor.rb:211:in `mon_synchronize'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/task.rb:159:in `invoke_with_call_chain'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/task.rb:187:in `block in invoke_prerequisites'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/task.rb:185:in `each'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/task.rb:185:in `invoke_prerequisites'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/task.rb:165:in `block in invoke_with_call_chain'
    from /Users/ahellesoy/.rvm/rubies/ruby-2.0.0-p0/lib/ruby/2.0.0/monitor.rb:211:in `mon_synchronize'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/task.rb:159:in `invoke_with_call_chain'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/task.rb:152:in `invoke'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/application.rb:143:in `invoke_task'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/application.rb:101:in `block (2 levels) in top_level'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/application.rb:101:in `each'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/application.rb:101:in `block in top_level'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/application.rb:110:in `run_with_threads'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0@global/gems/rake-10.0.3/lib/rake/application.rb:95:in `top_level'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0/gems/rake-compiler-0.8.3/bin/rake-compiler:24:in `<top (required)>'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0/bin/rake-compiler:23:in `load'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0/bin/rake-compiler:23:in `<main>'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0/bin/ruby_noexec_wrapper:14:in `eval'
    from /Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0/bin/ruby_noexec_wrapper:14:in `<main>'
rake aborted!
Command failed with status (1): [bundle exec rake-compiler cross-ruby VERSI...]
/Users/ahellesoy/github/bool/ruby/Rakefile:74:in `<top (required)>'
/Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0/bin/ruby_noexec_wrapper:14:in `eval'
/Users/ahellesoy/.rvm/gems/ruby-2.0.0-p0/bin/ruby_noexec_wrapper:14:in `<main>'
(See full trace by running task with --trace)
```

--- a/Makefile
+++ b/Makefile
@@ -42,8 +42,7 @@ mingw: mingw/bin/i686-w64-mingw32-gcc
 
 mingw/bin/i686-w64-mingw32-gcc:
 	mkdir -p mingw
-	# Don't attempt any of the newer versions - they don't work (gcc 4.7.0)
-	cd mingw && curl --silent --location http://downloads.sourceforge.net/project/mingw-w64/Toolchains%20targetting%20Win32/Automated%20Builds/mingw-w32-1.0-bin_i686-darwin_20110819.tar.bz2 | tar xvj
+	cd mingw && curl --silent --location http://downloads.sourceforge.net/project/mingw-w64/Toolchains%20targetting%20Win32/Automated%20Builds/mingw-w32-1.0-bin_i686-darwin_20120227.tar.bz2 | tar xvj
 endif
 
 ifeq ($(RUBY_PLATFORM), java)

